### PR TITLE
os/drivers: apply fb/lcd drivers structure

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1. porting the fb/lcd driver structure from nuttx
2. porting the ili9341 common driver
@gcjjyy

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3356e16b8e5fd6786ddec195e5f37e6f0cdf47ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3356e16b8e5fd6786ddec195e5f37e6f0cdf47ea] - Code Rule Check Result:
os/include/tinyara/fs/ioctl.h:313: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxV)
os/include/tinyara/fs/ioctl.h:314: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/include/tinyara/video/fb.h:134: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:135: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:137: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:140: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:142: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:145: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:147: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:171: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:174: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:176: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:178: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:180: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:183: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:186: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:188: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:189: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:191: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:193: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:195: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:265: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:266: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:267: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:282: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:283: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:285: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:286: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:288: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:289: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:291: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:292: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:294: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:295: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:298: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:299: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:301: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/tinyara/video/fb.h:302: ERROR: [IDT_M_TAB] code indent should use tabs where possible

**Comment by seinfra** [comment]
Target : [5ad98c71ffbca4fb46e3fba9ab2158991fac9136] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5ad98c71ffbca4fb46e3fba9ab2158991fac9136] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c79765f8e03fb8acfc4aed5c57acc9a377e4bb4b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c79765f8e03fb8acfc4aed5c57acc9a377e4bb4b] - Code Rule Check Result:
os/drivers/lcd/ili9341.c:134: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:134: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:135: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:135: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:171: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:171: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:198: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:198: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:199: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:199: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:200: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:200: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:201: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:201: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:202: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:202: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:226: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:226: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:227: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:227: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:228: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:229: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:230: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:230: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:253: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:253: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:254: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:254: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:255: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:255: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:256: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:256: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:257: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:257: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:258: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:258: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:283: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:283: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:284: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:284: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:285: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:285: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:286: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:286: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:287: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:287: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:305: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:305: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:317: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/ili9341.c:317: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/ili9341.c:550: ERROR: [IDT_M_TAB] please, no space before tabs

File: os/Kconfig.debug [hunk 1]
```diff
@@ -77,6 +77,38 @@ config DEBUG_AUDIO_INFO
...
+		Graphics Error Debug Output.
+
+config DEBUG_GRAPHICS_INFO
+	bool "Enable Graphics Informational Debug Output"
+	default n
+	depends on DEBUG_VERBOSE
+	---help---
+		Graphics Informational Debug Output.
+
+endif #DEBUG_LCD
```
Comment (ChinaTizenRT) [inline os/Kconfig.debug]: Line 80 using  "LCD" does not match here comment "DEBUG_LCD"

File: os/include/tinyara/input/touchscreen.h [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ */
+
+#ifndef __INCLUDE_NUTTX_INPUT_TOUCHSCREEN_H
+#define __INCLUDE_NUTTX_INPUT_TOUCHSCREEN_H
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+
+#include <nuttx/config.h>
```
Comment (ChinaTizenRT) [inline os/include/tinyara/input/touchscreen.h]: please remove all includes with component "nuttx"

File: os/include/tinyara/lcd/ili9341.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2018 Samsung Electronics All Rights Reserved.
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
```
Comment (ChinaTizenRT) [inline os/include/tinyara/lcd/ili9341.h]: why need change year here, is this newly add file?

File: os/drivers/lcd/ili9341.c [hunk 4]
```diff
@@ -0,0 +1,1318 @@
...
+ *   lcd       - Reference to private driver structure
+ *   x0        - Start x position
+ *   y0        - Start y position
+ *   x1        - End x position
+ *   y1        - End y position
+ *
+ ****************************************************************************/
+
+static void ili9341_selectarea(FAR struct ili9341_lcd_s *lcd,
+							   uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1)
```
Comment (ChinaTizenRT) [inline os/drivers/lcd/ili9341.c]: please check format here,  how to arrange more lines  parameter.

File: os/drivers/lcd/ili9341.c [hunk 5]
```diff
@@ -0,0 +1,1318 @@
...
+ *
+ ****************************************************************************/
+
+static int ili9341_getpower(FAR struct lcd_dev_s *dev)
+{
+	FAR struct ili9341_dev_s *priv = (FAR struct ili9341_dev_s *)dev;
+
+	if (priv) {
+		lcdvdbg("%d\n", priv->power);
+
```
Comment (ChinaTizenRT) [inline os/drivers/lcd/ili9341.c]: remove this blank line.

File: os/Kconfig.debug [hunk 6]
```diff
@@ -77,6 +77,38 @@ config DEBUG_AUDIO_INFO
 
 endif #DEBUG_AUDIO
 
+if LCD
```
Comment (xixidodo) [inline os/Kconfig.debug]: LGTM

**Comment by seinfra** [comment]
Target : [f0f5357d18455c500f3178ed028438ae11cee4f5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f0f5357d18455c500f3178ed028438ae11cee4f5] - Code Rule Check OK.

File: os/Kconfig.debug (hunk 1 above)
Comment (xiarihf) [reply os/Kconfig.debug]: LCD is not match the comment DEBUG_LCD,

File: os/board/stm32f429i-disco/src/stm32_lcd.c [hunk 7]
```diff
@@ -519,12 +519,15 @@ int up_lcdinitialize(void)
...
+ * Paramater:
+ *   display - Not used
+ *
  * Return:
  *   OK - On succes
  *
  ******************************************************************************/
 
-int up_fbinitialize(void)
+int up_fbinitialize(int display)
```
Comment (xiarihf) [inline os/board/stm32f429i-disco/src/stm32_lcd.c]: why to use useless parameter?

File: os/drivers/lcd/Kconfig [hunk 8]
```diff
@@ -0,0 +1,271 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
```
Comment (xiarihf) [inline os/drivers/lcd/Kconfig]: Change the Nuttx header here

File: os/drivers/lcd/Kconfig [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+
+config LCD_DMA_SUPPORT
+	bool "LCD support DMA transmit"
+	default n
+	---help---
+		Default putrun interface transmit data to LCD line by line, while some  
+		device may support DMA transmit, which could send data buffer of the update
+		area all together, to acheive a better flushing performance. to support this
+		config, the specific lcd driver need to implement corresponding interface
+
```
Comment (xiarihf) [inline os/drivers/lcd/Kconfig]: Only one empty line here please

File: os/drivers/lcd/ili9341.c [hunk 10]
```diff
@@ -0,0 +1,1318 @@
...
+	DEBUGASSERT(buffer && ((uintptr_t)buffer & 1) == 0);
+
+	/* Check if position outside of area */
+	if (col + npixels > ili9341_getxres(dev) || row > ili9341_getyres(dev)) {
+		return -EINVAL;
+	}
+
+	/* Select lcd driver */
+
```
Comment (xiarihf) [inline os/drivers/lcd/ili9341.c]: Too much empty lines in this file, please check it.

File: os/drivers/video/Kconfig [hunk 11]
```diff
@@ -0,0 +1,24 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
```
Comment (xiarihf) [inline os/drivers/video/Kconfig]: Change this nuttx header too.

File: os/include/tinyara/input/touchscreen.h (hunk 2 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/input/touchscreen.h]: fixed, thanks

File: os/Kconfig.debug (hunk 1 above)
Comment (SRCN-LiQin) [reply os/Kconfig.debug]: fixed, thanks

File: os/include/tinyara/lcd/ili9341.h (hunk 3 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/lcd/ili9341.h]: fixed, thanks

**Comment by seinfra** [comment]
Target : [6c133b56bef68baf7350c7efc73dc323dd2dae4e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6c133b56bef68baf7350c7efc73dc323dd2dae4e] - Code Rule Check OK.

File: os/drivers/lcd/ili9341.c (hunk 4 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/ili9341.c]: seems no need to modify

File: os/drivers/lcd/ili9341.c (hunk 5 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/ili9341.c]: blank line before return, seems ok

File: os/board/stm32f429i-disco/src/stm32_lcd.c (hunk 7 above)
Comment (SRCN-LiQin) [reply os/board/stm32f429i-disco/src/stm32_lcd.c]: follow the latest fb interface

File: os/drivers/lcd/Kconfig (hunk 8 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/Kconfig]: fixed, thanks

File: os/drivers/lcd/Kconfig (hunk 9 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/Kconfig]: fixed, thanks

File: os/drivers/video/Kconfig (hunk 11 above)
Comment (SRCN-LiQin) [reply os/drivers/video/Kconfig]: fixed, thanks

File: os/include/tinyara/video/fb.h [hunk 12]
```diff
@@ -70,130 +73,263 @@
...
+
+#ifdef CONFIG_FB_OVERLAY
+#define FBIOGET_OVERLAYINFO _FBIOC(0x0009)  /* Get video overlay info */
+/* Argument: writable struct
+ *           fb_overlayinfo_s */
+#define FBIO_SELECT_OVERLAY _FBIOC(0x000a)  /* Select overlay */
+/* Argument: read-only
+ *           unsigned long */
+#define FBIOSET_TRANSP      _FBIOC(0x000b)  /* Set opacity or transparency
+											* Argument: read-only struct
```
Comment (xiarihf) [inline os/include/tinyara/video/fb.h]: modify the alignment of this comment and below.

File: os/include/tinyara/input/touchscreen.h [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+ * convention, the touchscreen driver is registers as an input device at
+ * /dev/inputN where N uniquely identifies the driver instance.
+ *
+ * This header file documents the generic interface that all NuttX
+ * touchscreen devices must conform.  It adds standards and conventions on
+ * top of the standard character driver interface.
+ */
+
+#ifndef __INCLUDE_NUTTX_INPUT_TOUCHSCREEN_H
+#define __INCLUDE_NUTTX_INPUT_TOUCHSCREEN_H
```
Comment (sunghan-chang) [inline os/include/tinyara/input/touchscreen.h]: NUTTX->TINYARA

File: os/include/tinyara/input/touchscreen.h [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+#ifdef CONFIG_INPUT
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+
+/* IOCTL Commands *******************************************************************/
+/* Common TSC IOCTL commands */
+
+#define TSIOC_SETCALIB       _TSIOC(0x0001)  /* arg: Pointer to int calibration value */
```
Comment (sunghan-chang) [inline os/include/tinyara/input/touchscreen.h]: We don't have `_TSIOC`.

File: os/include/tinyara/input/touchscreen.h [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+#define __INCLUDE_NUTTX_INPUT_TOUCHSCREEN_H
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/fs/ioctl.h>
+
+#ifdef CONFIG_INPUT
```
Comment (sunghan-chang) [inline os/include/tinyara/input/touchscreen.h]: There is no INPUT config.

File: os/include/tinyara/lcd/ili9341.h [hunk 16]
```diff
@@ -320,3 +322,4 @@ int ili9341_clear(FAR struct lcd_dev_s *dev, uint16_t color);
 #endif
 
 #endif							/* __INCLUDE_TINYARA_LCD_ILI9341_H */
+
```
Comment (sunghan-chang) [inline os/include/tinyara/lcd/ili9341.h]: I don't know why you modified this file.

File: os/include/tinyara/video/fb.h [hunk 17]
```diff
@@ -1,6 +1,6 @@
 /****************************************************************************
  *
- * Copyright 2016 Samsung Electronics All Rights Reserved.
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline os/include/tinyara/video/fb.h]: 2016 is correct.

**Comment by sunghan-chang** [comment]
@SRCN-LiQin Could you resolve a conflict?

**Comment by seinfra** [comment]
Target : [e9197621ed221852646edf594d70ace3e24d2a29] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e9197621ed221852646edf594d70ace3e24d2a29] - Code Rule Check Result:
os/include/tinyara/fs/ioctl.h:270: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxV)
os/include/tinyara/fs/ioctl.h:271: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

**Comment by seinfra** [comment]
Target : [8219b469008f537a4931aad8b32ac2e797738c8e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8219b469008f537a4931aad8b32ac2e797738c8e] - Code Rule Check OK.

File: os/drivers/Makefile [hunk 18]
```diff
@@ -72,6 +72,8 @@ endif
 
 include analog$(DELIM)Make.defs
 include audio$(DELIM)Make.defs
+include video$(DELIM)Make.defs
+include lcd$(DELIM)Make.defs
```
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/drivers/Makefile]: fixed, thanks

File: os/include/debug.h [hunk 19]
```diff
@@ -882,6 +882,30 @@
 #define ullvdbg(...)
 #endif
 
+#ifdef CONFIG_DEBUG_LCD_ERROR
```
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/include/debug.h]: updated with alphabet order, thanks

File: os/include/debug.h [hunk 20]
```diff
@@ -1193,6 +1217,29 @@
 #define tmllvdbg    (void)
 #endif
 
+#ifdef CONFIG_DEBUG_LCD_ERROR
```
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/include/debug.h]: fixed, thanks

File: os/include/tinyara/fs/ioctl.h [hunk 21]
```diff
@@ -99,7 +99,8 @@
 #define _LWNLIOCBASE    (0x2400)	/* LWNL ioctl commands */
 #define _SECLINKBASE    (0x2500)	/* seclink ioctl commands */
 #define _IOTBUSBASE     (0x2600)	/* iotbus ioctl commands */
-#define _TESTIOCBASE (0xfe00)	/* KERNEL TEST DRV module ioctl commands */
+#define _FBIOCBASE      (0x2800)	/* Frame buffer character driver ioctl commands */
```
[the remark this answers is not in the document]
Comment (SRCN-LiQin) [reply os/include/tinyara/fs/ioctl.h]: yes it should be 0x2700, fixed, thanks

File: os/include/tinyara/input/touchscreen.h (hunk 13 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/input/touchscreen.h]: fixed similar issue in this file. thanks

File: os/include/tinyara/input/touchscreen.h (hunk 14 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/input/touchscreen.h]: added in fs/ioctl.h , thanks

File: os/include/tinyara/input/touchscreen.h (hunk 15 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/input/touchscreen.h]: this PR is mainly porting the lcd driver and related interfaces, while the INPUT config is expected to be added when we enable an input device

File: os/include/tinyara/video/fb.h (hunk 17 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/video/fb.h]: fixed, thanks

File: os/include/tinyara/lcd/ili9341.h (hunk 16 above)
Comment (SRCN-LiQin) [reply os/include/tinyara/lcd/ili9341.h]: yes there was only some coding rule modification in this file, for a uniform format with the latest fb/lcd interface

**Comment by seinfra** [comment]
Target : [1cc2dc182f4cf206060d2ffca49403d6c9a53084] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1cc2dc182f4cf206060d2ffca49403d6c9a53084] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [940790a88444bc80ffcc1e58e35671385daa189c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [940790a88444bc80ffcc1e58e35671385daa189c] - Code Rule Check OK.

File: os/drivers/lcd/Kconfig (hunk 8 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/Kconfig]: updated in latest commit, thanks

**Comment by SRCN-LiQin** [comment]
> @SRCN-LiQin Could you resolve a conflict?

resolved, thanks

**Pull request merged**